packages/components/calendar/src/index.vue

```diff
@@ -1,9 +1,7 @@
 <template>
   <div class="el-calendar">
     <div class="el-calendar__header">
-      <template v-if="$slots.header">
-        <slot name="header" :date="i18nDate"></slot>
-      </template>
+      <slot v-if="$slots.header" name="header" :date="i18nDate"></slot>
```

Comment:
I think you have completed misunderstood my comment.
1. You need to remove `v-if="$slots.header"` statement
2. You need to place the `v-else` part inside the `<slot>` tag so that when no slot is passed it can fallback to the content inside `<slot>`

Reply:
```
<slot name="header">This is default content</slot>
```
so IMO you doesn't need use `v-if`. Just put `v-else` part inside `slot` tag.

Reply:
I know your approach can also achieve the goal, but it is obviously better to use Vue's features directly.

Reply:
No problem, I'll change it